bugsnag-android-core/src/androidTest/java/com/bugsnag/android/ObserverInterfaceTest.java

```diff
@@ -50,59 +50,42 @@ public void tearDown() {
     @Test
     public void testAddMetadataToClientSendsMessage() {
         client.addMetadata("foo", "bar", "baz");
-        List<Object> metadataItem = (List<Object>)findMessageInQueue(
-                NativeInterface.MessageType.ADD_METADATA, List.class);
-        assertEquals(3, metadataItem.size());
-        assertEquals("foo", metadataItem.get(0));
-        assertEquals("bar", metadataItem.get(1));
-        assertEquals("baz", metadataItem.get(2));
+        StateEvent.AddMetadata msg = findMessageInQueue(StateEvent.AddMetadata.class);
+        assertEquals("foo", msg.getSection());
+        assertEquals("bar", msg.getKey());
+        assertEquals("baz", msg.getValue());
     }
 
     @Test
     public void testAddNullMetadataToClientSendsMessage() {
         client.addMetadata("foo", "bar", null);
-        List<Object> metadataItem = (List<Object>)findMessageInQueue(
-                NativeInterface.MessageType.REMOVE_METADATA, List.class);
-        assertEquals(2, metadataItem.size());
-        assertEquals("foo", metadataItem.get(0));
-        assertEquals("bar", metadataItem.get(1));
+        StateEvent.RemoveMetadata msg = findMessageInQueue(StateEvent.RemoveMetadata.class);
+        assertEquals("foo", msg.getSection());
+        assertEquals("bar", msg.getKey());
     }
 
     @Test
     public void testAddMetadataToMetadataSendsMessage() {
         client.addMetadata("foo", "bar", "baz");
-        List<Object> metadataItem = (List<Object>)findMessageInQueue(
-                NativeInterface.MessageType.ADD_METADATA, List.class);
-        assertEquals(3, metadataItem.size());
-        assertEquals("foo", metadataItem.get(0));
-        assertEquals("bar", metadataItem.get(1));
-        assertEquals("baz", metadataItem.get(2));
+        StateEvent.AddMetadata metadataItem = findMessageInQueue(StateEvent.AddMetadata.class);
+        assertEquals("foo", metadataItem.getSection());
+        assertEquals("bar", metadataItem.getKey());
+        assertEquals("baz", metadataItem.getValue());
     }
 
     @Test
     public void testClearTabFromClientSendsMessage() {
```

Comment:
We should also have a test for clearing metadata with key.

Reply:
I have added a case for `testAddNullMetadataSendsMessage` which I believe is what this is asked for here

Reply:
We also need one for `clearMetadata` with a key to generate a `RemoveMetadata` event.

Reply:
Got it - I've added a separate test for this case.